docs/source/manual/core.rst

```diff
@@ -26,28 +26,44 @@ Dropwizard consists mostly of glue code to automatically connect and configure t
 Organizing Your Project
 =======================
 
-In general, we recommend you separate your projects into three Maven modules: ``project-api``,
-``project-client``, and ``project-application``.
+If you plan on developing a client library for other developers to access your service, we recommend
+you separate your projects into three Maven modules: ``project-api``, ``project-client``, and
+``project-application``.
 
 ``project-api`` should contain your :ref:`man-core-representations`; ``project-client`` should use
 those classes and an :ref:`HTTP client <man-client>` to implement a full-fledged client for your
 application, and ``project-application`` should provide the actual application implementation, including
 :ref:`man-core-resources`.
 
-Our applications tend to look like this:
+To give a concrete example of this project structure, let's say we wanted to create a Stripe_-like
+API where clients can issue charges and the server would echo the charge back to the client.
+``stripe-api`` project would hold our ``Charge`` object as both the server and client want to work
+with the charge and to promote code reuse, ``Charge`` objects are stored in a shared module.
+``stripe-app`` is the Dropwizard application. ``stripe-client`` abstracts away the raw HTTP
+interactions and deserialization logic. Instead of using a HTTP client, users of ``stripe-client``
+would just pass in a ``Charge`` object to a function and behind the scenes, ``stripe-client`` will
+call the HTTP endpoint. The client library may also take care of connection pooling, and may
+provide a more friendly way of interpreting error messages. Basically, distributing a client library
+for your app will help other developers integrate more quickly with the service.
+
+If you are not planning on distributing a client library for developers, one
+can combine ``project-api`` and ``project-application`` into a single project,
+which tends to look like this:
 
 * ``com.example.myapplication``:
 
   * ``api``: :ref:`man-core-representations`.
   * ``cli``: :ref:`man-core-commands`
-  * ``client``: :ref:`Client <man-client>` implementation for your application
-  * ``core``: Domain implementation
+  * ``client``: :ref:`Client <man-client>` code used to access other HTTP endpoints -- not your endpoints.
```

Comment:
I'm going to throw this as it come up to my mind right now so feel free to ignore! :smile: 

Naming the module ``project-client`` (in the "client library" scenario) and the package ``client`` (in the "NO client library" scenario) _can_ still cause a little bit of misunderstanding (even though the new comments help a lot). 

Have you guys thought about renaming the ``client`` package into something like ``remote``, ``service`` or something like that?

To be more concrete, I'll write down an example based on the conventions I've adopted in one of my projects. I have two microservices, aggregation-service and recommendation-service. The aggregation-service depends on the recommendation-service in that it requires an invocation of the recommendation-service REST API. So, in the aggregation-service I have my ``remote`` package where I've created a JAX-RS annotated interface representing the recommendation-services and called ``RemoteRecommendationService`` (and ultimately instantiated using jersey-proxy-client). Again, this is just a (fairly opinionated) convention, but makes it clear what are the "remote" dependencies of a service and could potentially avoid the "client" naming clash...

Ultimately, that package (i.e. ``client``) should contain [_client_]s for [_external_]/[_remote_] [_service_]s (i.e. any word surrounded by square brackets is, kind of, representative of that package).